src/objects-table/ObjectsTable.js

```diff
@@ -260,14 +265,12 @@ class ObjectsTable extends React.Component {
             getValue: field.getValue || getFormatter(model, field.name),
         }));
 
-        const selectedHeaderChecked = !_.isEmpty(dataRows) && dataRows.every(row => row.selected);
+        const selectedHeaderChecked =
+            !_.isEmpty(dataRows) && dataRows.every(row => selection.includes(row.id));
 
         const allColumns = this.getColumns(columns, selectedHeaderChecked);
 
-        const activeRows = _(rows)
-            .keyBy("id")
-            .at(dataRows.filter(dr => dr.selected).map(dr => dr.id))
-            .value();
+        const activeRows = selection.map(id => rows.find(dr => dr.id === id));
```

Comment:
map -> find is O(n^2). To avoid this I usually resort to a keyBy + at, which is O(n)

Reply:
Also, about this.state.selection. Does it make sense to use a Set instead of an array? Check inclusions (`myset.has()`) are O(1) instead of O(n) this.